test/controllers/surveys_controller_test.rb

```diff
@@ -14,23 +14,45 @@ class SurveysControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "admin can see all surveys" do
-    assert false
+    create(:survey, user: @user)
+    create(:survey, user: @admin)
+
+    sign_in @admin
+    get surveys_url
+    assert_equal 2, @controller.instance_variable_get("@surveys").count
   end
 
   test "non admin only sees their own surveys" do
-    assert false
+    create(:survey, user: @user)
+    create(:survey, user: @admin)
+
+    sign_in @user
+    get surveys_url
+    assert_equal 1, @controller.instance_variable_get("@surveys").count
   end
 
   test "admin can edit anyones survey" do
-    assert false
+    @survey = create(:survey, user: @user)
+
+    sign_in @admin
+    get edit_survey_url(@survey)
+    assert_response :success
   end
 
   test "non admins can only edit their own surveys" do
-    assert false
+    @survey = create(:survey, user: @admin)
+
+    sign_in @user
+    get edit_survey_url(@survey)
+    assert_redirected_to root_path
   end
 
   test "show page for unpublished survey should redirect to root" do
-    assert false
+    @survey = create(:survey, user: @user)
```

Comment:
I feel like we should be explicit here, let's pass in `published: false` just to be safe, we might end up making the factory default to true in the future

Reply:
Ok @spencerldixon I will fix this.